core/payment-driver/glmsync/src/driver.rs

```diff
@@ -323,13 +323,15 @@ Mind that to be eligible you have to run your app at least once on testnet -
         msg: ValidateAllocation,
     ) -> Result<bool, GenericError> {
         let (network, _) = platform_to_network_token(msg.platform)?;
+        let tx_fee_cost = wallet::get_tx_fee(&msg.address, network).await?;
```

Comment:
please note `get_tx_fee(` takes the "recipient" as an argument, but you are passing the "sender". ( this is correct, the real recipient is not known yet )

There is a difference in fees when transferring to a new vs existing account, existing is cheaper.
This means this check always fetches the cheaper fee ( best case scenario )

Maybe worth to add a comment about this.